colcon_powershell/shell/template/prefix.ps1.em

```diff
@@ -15,11 +15,15 @@ if ($env:COLCON_PYTHON_EXECUTABLE) {
   $_colcon_python_executable="@(python_executable)"
   # if it doesn't exist try a fall back
   if (!(Test-Path "$_colcon_python_executable" -PathType Leaf)) {
-    if (!(Get-Command "python" -ErrorAction SilentlyContinue)) {
-      echo "error: unable to find Python executable"
+@{
+import sys
+python_fallback_executable = 'python' if sys.platform == 'win32' else 'python3'
```

Comment:
> Should this use a logic of asking env on non Windows platforms like we do in the other shells ?

Can you point to the specific logic you have in mind.

> we just need to be aware that this will fail on platforms that use "python" for python3 and don't have a python3 executable

Indeed, such a system would also "violate" the Python guidelines though.

Reply:
nevermind I was thinking of something like `/usr/bin/env python3` or `which python3` but in both case it'll points to the first executable on the path so equivalent to `python3`.

> Indeed, such a system would also "violate" the Python guidelines though.

Yeah I guess systems that were doing so did backpedal since. Apparently arch provide both python and python3 pointing to the same bin now